src/hotspot/share/interpreter/interpreterRuntime.cpp

```diff
@@ -190,7 +190,7 @@ JRT_ENTRY(void, InterpreterRuntime::resolve_ldc(JavaThread* current, Bytecodes::
     // The bytecode wrappers aren't GC-safe so construct a new one
     Bytecode_loadconstant ldc2(m, last_frame.bci());
     int rindex = ldc2.cache_index();
-    if (rindex < 0)
+    if (rindex < 0 && m->constants()->resolved_references() != NULL)
```

Comment:
Should be `resolved_references_or_null()`

Reply:
I think the cache should always be initialized. The problem are the resolved symbols in the cache. But I don't see any harm in adding this check as well.